pkg/odo/cli/component/exec.go

```diff
@@ -0,0 +1,115 @@
+package component
+
+import (
+	"fmt"
+	"github.com/openshift/odo/pkg/envinfo"
+	projectCmd "github.com/openshift/odo/pkg/odo/cli/project"
+	"github.com/openshift/odo/pkg/odo/genericclioptions"
+	odoutil "github.com/openshift/odo/pkg/odo/util"
+	"github.com/openshift/odo/pkg/odo/util/completion"
+	"github.com/openshift/odo/pkg/odo/util/experimental"
+	"github.com/openshift/odo/pkg/odo/util/pushtarget"
+	"github.com/openshift/odo/pkg/util"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	ktemplates "k8s.io/kubectl/pkg/util/templates"
+	"path/filepath"
+)
+
+// ExecRecommendedCommandName is the recommended exec command name
+const ExecRecommendedCommandName = "exec"
+
+var execExample = ktemplates.Examples(`  # Executes a command inside the component
+%[1]s
+`)
+
+// ExecOptions contains exec options
+type ExecOptions struct {
+	componentContext string
+	componentOptions *ComponentOptions
+	devfilePath      string
+	namespace        string
+
+	command         []string
+	EnvSpecificInfo *envinfo.EnvSpecificInfo
```

Comment:
why additionally using `EnvSpecificInfo`? we have it in `context` inside `ComponentOptions`?
(not specific for your PR, it is duplicate at other places as well)

Reply:
Fixed